hack/build/push_image.sh

```diff
@@ -13,10 +13,18 @@ out_image="${IMG:-quay.io/dynatrace/dynatrace-operator}:${TAG}"
 
 # directory required by docker copy command
 mkdir -p third_party_licenses
-docker build . -f ./Dockerfile -t "${base_image}" \
+
+if [[ "${NO_CACHE}" ]]; then
```

Comment:
is there ever a reason to run this without cache?
if yes, would it make sense to split `push_image` into `build_image` and `push_image`, with additional parameters possible for `push_image` (not just 1 very specific one)?

Reply:
undoing this change, and making this nicer in a different PR